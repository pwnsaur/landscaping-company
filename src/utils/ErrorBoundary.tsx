import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
