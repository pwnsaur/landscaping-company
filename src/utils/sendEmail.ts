import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
