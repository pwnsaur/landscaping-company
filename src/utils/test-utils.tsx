import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
