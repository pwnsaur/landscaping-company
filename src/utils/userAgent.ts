export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
