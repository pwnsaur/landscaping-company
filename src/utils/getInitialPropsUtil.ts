import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
