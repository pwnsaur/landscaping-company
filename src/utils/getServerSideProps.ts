import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
