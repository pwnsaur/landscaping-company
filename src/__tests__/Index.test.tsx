import Home from '@/pages/index';
import { render } from '@/utils/test-utils';

describe('Home Page', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
