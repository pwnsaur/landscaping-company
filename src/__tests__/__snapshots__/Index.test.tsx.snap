// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches the snapshot 1`] = `
<DocumentFragment>
  .c5 {
  min-width: 9rem;
  height: 3rem;
  padding: 0 1rem;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 1rem;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #ededed;
  border: 2px solid #21412a;
  color: #21412a;
}

.c5:hover {
  background-color: #21412a;
  border-color: #ededed;
  color: #ededed;
}

.c0 {
  position: relative;
  background-image: url(/img.jpg);
  background-attachment: fixed;
  background-size: cover;
  background-repeat: repeat-y;
  background-position: top center;
  width: 100%;
  min-height: 100vh;
  margin-bottom: -5rem;
}

.c1 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: calc(100vh - 141px);
}

.c2 {
  font-size: 4rem;
  text-align: center;
  color: white;
  text-transform: uppercase;
}

.c3 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  margin-bottom: 20rem;
  background-color: #ededed;
}

.c4 {
  display: block;
  width: 70%;
  padding: 2rem 6rem;
  font-size: 1.2rem;
  line-height: 1.6;
  text-align: justify;
}

<div
    class="c0"
    id="parallax"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        The Company
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur, officiis. Illo assumenda velit ratione enim, quaerat inventore nostrum sequi vel nihil, tempore maxime ad. Quibusdam rerum ducimus voluptates provident sit?
      </div>
      <a
        href="/projects"
      >
        <button
          class="c5"
        >
          Projekti
        </button>
      </a>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Odit itaque dolorum minus quisquam sint sunt ad sit vero, eos repudiandae in doloribus est porro veniam enim fugiat ipsam laborum at?
      </div>
      <a
        href="/services"
      >
        <button
          class="c5"
        >
          Pakalpojumi
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
